File: practise/leetcode/DailyQuestion/202406/575.ts
function distributeCandies(candyType: number[]): number {
    let s = new Set(candyType);
    return Math.min(s.size,candyType.length/2);
};

function distributeCandie(candyType: number[]): number {
    let ctmp:{[key:number]:number}={};
    for(let e of candyType){
        if(e in ctmp){
            ctmp[e]++;
        }else{
            ctmp[e]=1;
        }
    }
    let ans=Math.floor(candyType.length/2); // floor表示向下取整
    return Math.min(ans,Object.keys(ctmp).length);
};

File: practise/leetcode/DailyQuestion/202406/575.go
func distributeCandies(candyType []int) int {
	ctmp := make(map[int]int)
	for _, e := range candyType {
		ctmp[e]++
	}
	ans := len(candyType) / 2
	if len(ctmp) < ans {
		return len(ctmp)
	}
	return ans
}

func distributeCandies(candyType []int) int {
    set := map[int]struct{}{}
    for _, t := range candyType {
        set[t] = struct{}{}
    }
    return min(len(set), len(candyType)/2)
}
